### PR TITLE
feat(firestore-stripe-payments): allow to be specify trial period days

### LINKED ISSUE #599: Support trial — there's no longer the legacy option for setting a trial in Stripe
@jsteele-stripe  There's no longer the legacy option for setting a trial in Stripe. So `trial_from_plan` workaround is no longer valid.

_Originally posted by @alex-i0 in https://github.com/invertase/stripe-firebase-extensions/issues/274#issuecomment-1875938597_

### PR DESCRIPTION
**Pull request by jauntybrain**
Allows to specify number of trial period days before initiating checkout flow. 
Does not depend on the now-legacy `trial_from_plan` parameter.

Fixes #599, #274

### CODE REVIEW

File: firestore-stripe-payments/functions/src/index.ts [hunk 1]
```diff
@@ -209,9 +210,12 @@ exports.createCheckoutSession = functions
           sessionCreateParams.payment_method_collection =
             payment_method_collection;
           sessionCreateParams.subscription_data = {
-            trial_from_plan,
             metadata,
           };
+          if (trial_period_days) {
+            sessionCreateParams.subscription_data.trial_period_days =
```
Comment (pr-Mais) [inline firestore-stripe-payments/functions/src/index.ts:216]: any possible undefined values here?

**Review by pr-Mais — APPROVED** [review]
LGTM

File: firestore-stripe-payments/functions/src/index.ts (hunk 1 above)
Comment (jauntybrain) [reply firestore-stripe-payments/functions/src/index.ts:216]: Nope, should be all good - `subscription_data` always exists (we set it on L212) and we check for `trial_period_days` being null.

**Comment by LimarenkoDenis** [comment]
@jauntybrain Thank you! 

I was waiting for this fix

**Comment by carloshuff1** [comment]
Potentially a noob problem, but when I specify trial_period_days its not taking me to a trial page. It is taking me to a regular subscription page without the subscription days added. Don't understand what i'm doing wrong. 

Invertase stripe firebase extension. Firebase v9.

SNIPPET: 
document.getElementById('start-free-trial').addEventListener('click', async () => {
            try {
                const checkoutSessionRef = collection(db, 'customers', user.uid, 'checkout_sessions');
                const docRef = await addDoc(checkoutSessionRef, {
                    price: 'price_fake1OdlyKKhmfO9fl4z8ZX65Qma',
                    trial_period_days: 7,
                    success_url: window.location.origin,
                    cancel_url: window.location.origin
                });

**Comment by jacob-i** [comment]
Thanks for the fix!

**Comment by YoLoADR** [comment]
> trial_from_plan

Thanks @carloshuff1  carloshuff1

**Pull request merged**